source/codegen/templates/service_helpers.mako

```diff
@@ -282,23 +296,31 @@ ${initialize_standard_input_param(function_name, parameter)}\
   parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
 %>\
 %   if common_helpers.is_enum(parameter) == True:
-%     if enums[parameter["enum"]].get("generate-mappings", False):
 <%
-  map_name = parameter["enum"].lower() + "_output_map_"
-  mapped_enum_iterator_name = parameter_name + "_omap_it"
+  has_mapped_enum = 'mapped-enum' in parameter
+  has_unmapped_enum = 'enum' in parameter
+  if has_mapped_enum:
+    mapped_enum_name = parameter["mapped-enum"]
+    mapped_enum_type = mapped_enum_name[:-len("Mapped")] if mapped_enum_name.endswith("Mapped") else mapped_enum_name
+    map_name = mapped_enum_type.lower() + "_output_map_"
+    mapped_enum_iterator_name = parameter_name + "_omap_it"
 %>\
+%     if has_mapped_enum:
         auto ${mapped_enum_iterator_name} = ${map_name}.find(${parameter_name});
         if(${mapped_enum_iterator_name} != ${map_name}.end()) {
-          response->set_${parameter_name}(static_cast<${namespace_prefix}${parameter["enum"]}>(${mapped_enum_iterator_name}->second));
+          response->set_${parameter_name}_mapped(static_cast<${namespace_prefix}${mapped_enum_type}Mapped>(${mapped_enum_iterator_name}->second));
```

Comment:
Can this be the type in the map instead of a `static_cast`?

Reply:
We probably could add it to map. We might need to write some mutation or a lookup helper so that this type information is available where we declare maps. Let me see what it takes to get the value there.

Although, even though map's type would have an enum name, we might still have to put integer constants there instead of using enum constants names for better readability of map. Otherwise the map declaration will have to be multi-line with (enum_name, raw_value) pair on each line.

Reply:
That's fine. I'd like less `cast`s overall. But if it's not an easy fix here, you can leave it as is.

Reply:
Turns out getting enum type in the map was quite easy, but following code doesn't compile since integer values in the map don't match with enum type specified in the map type. Looks like we can't make that change unless we add enum values in the map itself, which will make it unreadable.

>   std::map<nidevice_grpc::MobileOSNames, std::string> mobileosnames_input_map_ { {1, "Android"},{2, "iOS"},{3, "None"}, };

I am leaving the casts as is.